README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# crdata

<!-- badges: end -->

The goal of {crdata} is to provide easy access to data I’ve used in
research and teaching, and its documentation.

## Minimal Example

Here’s a minimal example to illustrate the recommended use.

``` r
# install package (once per computer)
# remotes::install_github("carlislerainey/crdata")

# load data (once per session)
weis <- crdata::weisiger2014

# open documentation (as needed)
?crdata::weisiger2014
```

## Installing the Package

To install the development version from
[GitHub](https://github.com/carlislerainey/crdata) use:

    # install latest crdata package from github
    # install.packages("remotes")
    remotes::install_github("carlislerainey/crdata")

I have no plans at the moment to upload to CRAN.

## Data Sets

| Data Set       | References                                                                                                                                                                                                                                   | Topics                                                                                  |
|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------|
| `bm2015`       | [Rauchhaus(2009)](https://doi.org/10.1177/0022002708330387); [Bell and Miller (2015)](https://doi.org/10.1177/0022002713499718); [Rainey (2016)](https://doi.org/10.1093/pan/mpw014); [Rainey (2023)](https://doi.org/10.31235/osf.io/bmvnu) | Logistic regression with separation in large samples.                                   |
| `br2014`       | [Barrilleaux and Rainey (2014)](https://doi.org/10.1177/1532440014561644); [Rainey (2016)](https://doi.org/10.1093/pan/mpw014); [Rainey (2023)](https://doi.org/10.31235/osf.io/bmvnu)                                                       | Logistic regression with separation in small samples.                                   |
| `weisiger2014` | [Weisiger (2014)](https://journals.sagepub.com/doi/10.1177/0738894213508691); [Rainey and McCaskey (2021)](https://doi.org/10.1017/psrm.2021.9)                                                                                              | Logistic regression in small samples; Firth’s logit; R packages {brglm2} and {logistf}. |

## Using the Data

There are three ways to access the data. I’ll illustrate the syntax with
the `weisiger2014` data set.

1.  **Recommended**: First, you can use `weis <- crdata::weisiger2014`.
    I prefer this approach because it mirrors `weis <- read_csv(path)`
    most closely.
2.  Second, you can refer to the data set with the `::` operators
    without creating an object. Anytime you want to use the
    `weisiger2014` data set, simply use `crdata::weisiger2014`.
3.  Third, you can run `library(crdata)`, which lazily loads all the
    data sets, so that `weisiger2014` (and all others) are available
    without the `crdata::` prefix.

To see the documentation, run `?crdata::weisiger2014`.

## Example

For example, McCaskey and Rainey (2021) uses data from Weisiger (2014)
to illustrate Firth’s logit using a binary outcome in small samples.
Here’s a comparison of logit models fit using maximum likelihood and
Firth’s penalized maximum likelihood.

``` r
# load packages
library(tidyverse)

# load data
weis <- crdata::weisiger2014

# rescale weisiger2014 explanatory variables using arm::rescale()
rs_weis <- weis %>%
  mutate(across(polity_conq:coord, arm::rescale)) 

# fit models
f <- resist ~ polity_conq + lndist + terrain + soldperterr + gdppc2 + coord
ls <- lm(f, data = rs_weis) # lpm (replicates Table 2 Model 3 of Weisiger 2014)
mle <- glm(f, data = rs_weis, family = "binomial") # logistic regression
pmle <- glm(f, data = rs_weis, family = "binomial", method = brglm2::brglmFit) # logistic regression


# plot coefficients
modelsummary::modelplot(list("Logit via MLE" = mle, 
                             "Logit via Firth's PMLE" = pmle), 
                        conf_level = 0.90)
```

<img src="man/figures/README-setup-1.png" width="75%" style="display: block; margin: auto;" />

## Citation

Please cite the authors of the original data (both the data set and the
article)! The appropriate references to papers and data sets are given
in the help file for each data set (e.g., `?crdata::weisiger2014`).


NEWS.md
# crdata 0.0.0.9000

Pre-release version for testing.


R/bm2015.R
#' A data set on nuclear weapons and war; illustrates logistic regression with separation in a large sample
#'
#' Bell and Miller (2015) data set used in Rainey (2016) and Rainey (2023) to illustrate logistic regression models with separation. These are the data to reproduce the "Firth logit" results with "Kargil excluded" throughout Bell and Miller (2015). See the "Firth logit" column of Table B of their online appendix.
#'
#' @format A data frame with 455,619 observations and 14 variables:
#' \describe{
#' \item{\code{year}}{the year of the observation (i.e., dyad-year)}
#' \item{\code{statea}}{the numeric COW code of the first state of the dyad}
#' \item{\code{stateb}}{the numeric COW code of the second state of the dyad}
#' \item{\code{warl2}}{whether a war occurred}
#' \item{\code{onenukedyad}}{whether exactly one state has nuclear weapons}
#' \item{\code{twonukedyad}}{whether both states have nuclear weapons}
#' \item{\code{logCapabilityRatio}}{the natural log of the capability ratio, which measures the distribution of power between the two states}
#' \item{\code{Ally}}{whether the two states have a formal alliance or nonaggression pact}
#' \item{\code{SmlDemocracy}}{the smaller of the two Polity scores in the dyad}
#' \item{\code{SmlDependence}}{measures economic interdependence; the smaller of each country's imports and exports with its partner divided by their GDP}
#' \item{\code{logDistance}}{the natural log of the distance between capitals of two states, or for large states, the distance between nearest ports}
#' \item{\code{Contiguity}}{whether the two states are continguous}
#' \item{\code{MajorPower}}{whether at least one of the states is a major power}
#' \item{\code{NIGOs}}{measures shared membership in intergovernmental organizations}
#' }
#' For further details, see Rauchhaus (2009, pp. 266-268) and Pevehouse and Russett (2006, pp. 980-983).
#'
#'
#' @references
#'
#' Bell, Mark S., and Nicholas L. Miller. 2013.
#' "Questioning the Effect of Nuclear Weapons on Conflict."
#' \emph{Journal of Conflict Resolution} 59(1): 74–92. \doi{10.1177/0022002713499718}.
#'
#' Pevehouse, Jon, and Bruce Russett.
#' 2006.
#' "Democratic International Governmental Organizations Promote Peace."
#' \emph{International Organization} 60(4): 969-1000.
#' \doi{10.1017/S0020818306060322}.
#'
#'
#' Rainey, Carlisle. 2016.
#' "Dealing with Separation in Logistic Regression Models." \emph{Political Analysis} 24(3): 339-355. \doi{10.1093/pan/mpw014}.
#'
#'  Rainey, Carlisle. 2023.
#' "Hypothesis Tests Under Separation." Forthcoming in \emph{Political Analysis}. \doi{10.31235/osf.io/bmvnu}.
#'
#' Rainey, Carlisle.
#' 2016.
#' "priors-for-separation.zip"
#' Replication Data for: Dealing with Separation in Logistic Regression Models. Harvard Dataverse, V1. \doi{10.7910/DVN/VW7G2Q/MTJB9H}.
#'
#' Rauchhaus, Robert. 2009. "Evaluating the Nuclear Peace Hypothesis."
#' \emph{Journal of Conflict Resolution} 53(2): 258–77. \doi{10.1177/0022002708330387}.
#'
#'
#' @examples
#'
#' # a simple example
#'
#' bm <- crdata::bm2015
#'
#' # formula to reproduce "Firth logit" column of Table B of Bell and Miller's online appendix.
#' f <- warl2 ~ onenukedyad + twonukedyad + logCapabilityRatio +
#'   Ally + SmlDemocracy + SmlDependence + logDistance +
#'   Contiguity + MajorPower + NIGOs
#'
#' # twonukedyad == 1 perfectly predicts warl2 == 0
#' table("warl2" = bm$warl2, "twonukedyad" = bm$twonukedyad)
#'
#' # logit model with separation
#' fit <- glm(f, data = bm, family = binomial)
#'
#' # Rainey (2023) shows that Wald tests can never reject null when variables
#' #   create separation
#' summary(fit)
#'
#' # Rainey (2023) shows that the LR test works fine when variables create
#' #   separation, though.
#' fit0 <- update(fit, . ~ . - twonukedyad)
#' anova(fit0, fit, test = "Chisq")
#'
#'
"bm2015"
